### ISSUE TITLE
BUG: Series([big_timestamp], dtype=pd.DatetimeTZDtype("us", "US/Pacific")) raises

### ISSUE THREAD

**Issue by jbrockmendel** [pandas-dev/pandas#54442]
```python
ts = pd.Timestamp(2999, 1, 1)
ts2 = ts.tz_localize("US/Pacific")

dtype = pd.DatetimeTZDtype("us", "US/Pacific")

>>> ser = pd.Series([ts2], dtype=dtype)
[...]
pandas._libs.tslibs.np_datetime.OutOfBoundsDatetime: Cannot cast 2999-01-01 00:00:00-08:00 to unit='ns' without overflow.
```

**Comment by AdrianDAlessandro** [comment]
take

**Comment by AdrianDAlessandro** [comment]
Further investigation reproduces this bug in multiple ways. It appears the underlying Cython code (in `pandas._libs.tslib.pyx`) still requires ns resolution. This is not limited to timezone-aware datetimes.

```python
>>> pd.Series([pd.Timestamp(2999,1,1)], dtype="datetime64[us]")
[...]
pandas._libs.tslibs.np_datetime.OutOfBoundsDatetime: Cannot cast 2999-01-01 00:00:00 to unit='ns' without overflow., at position 0
```

Also if you don't specify the dtype for an out-of-bounds datetime then you just get an array with object dtype, and it doesn't infer it.
```python
>>> pd.Series([pd.Timestamp(2000,1,1)])
0   2000-01-01
dtype: datetime64[ns]
>>> pd.Series([pd.Timestamp(2999,1,1)])
0    2999-01-01 00:00:00
dtype: object
```

**Comment by jbrockmendel** [comment]
closed by #55768
